src/analyzer/protocol/dns/DNS.cc

```diff
@@ -628,6 +631,50 @@ bool DNS_Interpreter::ParseRR_SRV(detail::DNS_MsgInfo* msg, const u_char*& data,
     return true;
 }
 
+bool DNS_Interpreter::ParseRR_NAPTR(detail::DNS_MsgInfo* msg, const u_char*& data, int& len, int rdlength,
+                                    const u_char* msg_start) {
+    zeek_uint_t order = ExtractShort(data, len);
+    zeek_uint_t preference = ExtractShort(data, len);
+    rdlength -= 4;
+
+    if ( len <= 0 || rdlength <= 0 ) {
+        analyzer->AnalyzerViolation("DNS_NAPTR_too_short");
+        return false;
+    }
+
+    // These all check rdlength and return nullptr if there's not enough data available.
+    auto flags = extract_char_string(analyzer, data, len, rdlength);
+    auto service = extract_char_string(analyzer, data, len, rdlength);
+    auto regexp = extract_char_string(analyzer, data, len, rdlength);
+
+    // The replacement string is a name. Compression shouldn't be used, but doesn't seem
+    // we have a helper that would allow to control this.
+    u_char replacement[513];
```

Comment:
Drive-by: Should we define this length as a constant somewhere since we reuse it a bunch of times in this file?